docs/nnue.md

```diff
@@ -1361,8 +1362,8 @@ int32_t* linear_sparse_input(
 
             m256_process_chunk(sum0, sum1, col0, _mm256_setzero_si256(), factor);
 
-            _mm256_store_si256(output + output_offset0, sum0);
-            _mm256_store_si256(output + output_offset1, sum1);
+            _mm256_store_si256(output + output_offset0, _mm256_srai_epi32(sum0, log2_weight_scale));
```

Comment:
While there does indeed appear to be an omission regarding the output scaling, I believe this is wrong. Since this is inside `for (; i < num_nnz_input_indices; ++i) {` the same outputs can be hit multiple times - resulting in a recursive process similar to decaying average. Also, the scaling has to be applied to the main loop above too. Needs to be moved outside of these loops overall.

Reply:
Yes that makes sense. So how would you fix this? I was thinking of reverting this change and adding something like this to the end of the function
```c
for (int i = 0; i < layer.num_outputs; i += output_register_width) {
    _mm256_store_si256(output + i, _mm256_srai_epi32(_mm256_load_si256(output + i), log2_weight_scale));
}
```
Alternatively I could just revert these changes if you prefer. These two functions in particular are the only ones I haven't tried implementing myself, so I can't actually test this code.

Reply:
That's the solution I had in mind. I think the change is important because otherwise the layer calculation is not complete.

Reply:
Ok, I've added the changes.